code/game/machinery/computer/medical.dm

```diff
@@ -55,19 +52,66 @@
 <BR><A href='?src=[REF(src)];logout=1'>{Log Out}</A><BR>
 "}
 				if(2)
+					// JS scripts for the filter
+					dat += {"
+					<script src="[SSassets.transport.get_asset_url("jquery.min.js")]"></script>
+					<script type='text/javascript'>
+
+						function updateSearch(){
+							var filter_text = document.getElementById('filter');
+							var filter = filter_text.value.toLowerCase();
+
+							if(complete_list != null && complete_list != ""){
+								var mtbl = document.getElementById("maintable_data_archive");
+								mtbl.innerHTML = complete_list;
```

Comment:
I don't know how significant this is, but it seems horribly inefficient, since I think this has to parse the whole table's HTML every time the search updates, which is on every keystroke. Wouldn't it be better *and* cleaner to instead call `show` on every matching element and `hide` on the rest?

Reply:
Maybe I should've specified - I mean removing the restoration of `innerHTML` and instead adding `show` as an alternative where you're `hide`-ing elements.

But also, that depends on if you consider an empty string to match everything :sunglasses: